controllers/clusterenvironment_controller.go

```diff
@@ -33,6 +39,7 @@ type ClusterEnvironmentReconciler struct {
 	Scheme *runtime.Scheme
 }
 
+//+kubebuilder:rbac:groups="",resources=secrets,verbs=get;list;watch
```

Comment:
From what I can tell, we don't need `list` or `watch` permissions on secrets.
```suggestion
//+kubebuilder:rbac:groups="",resources=secrets,verbs=get
```

Reply:
list is required, watch can be removed